File: iOS/Spaces V1.5/Models/EncryptedMediaMetadata.swift
import Foundation

struct EncryptedMediaMetadata: Hashable {
    let mediaId: String
    let mediaType: MediaType
    let storagePath: String
    let thumbnailStoragePath: String?
    let encryptionVersion: String
    let nonce: String
    let thumbnailNonce: String?
    let mimeType: String
    let fileSize: Int
    let width: Int?
    let height: Int?
    let duration: Double?
    let createdAt: Date?
    let uploadedBy: String
}

File: Android/app/src/main/java/com/arcinteractive/spaces/data/model/EncryptedMediaMetadata.kt
package com.arcinteractive.spaces.data.model

import java.util.Date

data class EncryptedMediaMetadata(
    val mediaId: String,
    val mediaType: MediaType,
    val storagePath: String,
    val thumbnailStoragePath: String? = null,
    val encryptionVersion: String,
    val nonce: String,
    val thumbnailNonce: String? = null,
    val mimeType: String,
    val fileSize: Long,
    val width: Int? = null,
    val height: Int? = null,
    val duration: Double? = null,
    val createdAt: Date? = null,
    val uploadedBy: String
)
